File: 11_Container_With_Most_Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxVolume = 0
        left = 0
        right = len(height) - 1

        while left < right:
            x = right - left
            h = min(height[left], height[right])
            maxVolume = max(maxVolume, x*h)

            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        
        return maxVolume

File: swift/11_container_with_most_water.swift
class Solution {
  func maxArea(_ height: [Int]) -> Int {
    var left: Int = 0
    var right: Int = height.count - 1
    var curMaxArea: Int = 0

    while left < right {
      let leftHeight: Int = height[left]
      let rightHeight: Int = height[right]
      let minHeight: Int = min(leftHeight, rightHeight)

      curMaxArea = max(curMaxArea, minHeight * (right - left))

      if leftHeight > rightHeight {
        right -= 1
      } else {
        left += 1
      }
    }

    return curMaxArea
  }
}
